### PR TITLE
move `is_coinbase_first` check to consensus

### LINKED ISSUE #907: Move `is_coinbase_first` check out of `zebra_chain` and into `zebra_consensus`.
This is a semantic validation check, so it does not belong in `zebra_chain`.  Ideally we could write all of the consensus checks as functions in, e.g., `zebra_consensus::block`, then call them all in sequence.

### PR DESCRIPTION
**Pull request by oxarbitrage**
Attempt to close https://github.com/ZcashFoundation/zebra/issues/907 by adding a new module `consensus::block::checks` with only `is_coinbase_first()`function by now.

### CODE REVIEW

File: zebra-consensus/src/block.rs
Comment (hdevalence) [inline zebra-consensus/src/block.rs]:
```suggestion
```

It's good to avoid glob imports like this unless they're really necessary, because it makes it harder to keep track of what items are in the local namespace.  Instead of doing a glob import, if the `checks` module is renamed to `check`, then the code below could be

```
check::is_coinbase_first(&block);
```
which is nice because the two parts of the name fit together.  Then when there are more checks, we can read off the code as a sequence of instructions:
```
check::is_coinbase_first(&block)?;
check::something_else(...)?;
check::another_property(...)?;
```

File: zebra-consensus/src/block.rs
Comment (hdevalence) [inline zebra-consensus/src/block.rs:110]: I didn't spot it when making the issue that this PR closes but I think that this function should also move, for the same reason.  Moving it doesn't have to be in this PR though, it could be in a later one.

**Review by hdevalence — CHANGES_REQUESTED** [review]
This looks great.  Before merging I think it should try to avoid using glob imports (more details in the review comment).

File: zebra-consensus/src/block.rs
Comment (oxarbitrage) [reply zebra-consensus/src/block.rs]: Nice one, applied at https://github.com/ZcashFoundation/zebra/pull/917/commits/9df6cf2b6da36979634cbc0a1d2e74f5d0272a8b

File: zebra-consensus/src/block.rs
Comment (oxarbitrage) [reply zebra-consensus/src/block.rs:110]: The problem with that one is that it used inside `zebra_chain`. It means we will have to depend on consensus at chain, i have the strong impression we will not want to do that.

**Review by hdevalence — APPROVED** [review]
LGTM!

File: zebra-consensus/src/block.rs
Comment (hdevalence) [reply zebra-consensus/src/block.rs:110]: Ah yeah, that's worth digging in to, let's not block this change on adding that one.

**Pull request merged**
